### PR TITLE
[BC-breaking] Split raw text classification datasets into individual files

### PR DESCRIPTION
**Pull request by cpuhrsch**
To further align the dataset organization of the raw datasets with the other domain libraries (torchaudio and torchvision) we split them into separate files, one each per dataset. This PR introduces this change for all text classification datasets. It is BC-breaking in the sense that the torchtext.experimental.datasets.raw.text_classification module is being removed.

Follow-up:
- Split raw sequence tagging datasets into individual files
- Split raw translation datasets into individual files
- Split raw language modeling datasets into individual files
- Split raw question answer datasets into individual files

TODO:
- [x] Add unit test, one per dataset, that compares first line output and verifies dataset constants.

### CODE REVIEW

File: torchtext/experimental/datasets/raw/amazonreviewpolarity.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    'test': 400000,
+}
+
+_PATH = 'amazon_review_polarity_csv.tar.gz'
+
+
+@wrap_split_argument
+@add_docstring_header()
+def AmazonReviewPolarity(root='.data', split=('train', 'test'), offset=0):
+    def _create_data_from_csv(data_path):
```
Comment (zhangguanheng66) [inline torchtext/experimental/datasets/raw/amazonreviewpolarity.py:24]: I feel we don't need to repeat `_create_data_from_csv` function for all the datasets. We can have a utility func and keep it private somewhere.

**Review by zhangguanheng66 — COMMENTED** [review]
I thought we don't use [BC-breaking] for the prototype features.
And users will experience BC breaking if they import the datasets from `torchtext.experimental.datasets.raw.text_classification`, right?

File: torchtext/experimental/datasets/raw/amazonreviewpolarity.py (hunk 1 above)
Comment (cpuhrsch) [reply torchtext/experimental/datasets/raw/amazonreviewpolarity.py:24]: Yes, I agree. I'll think of some generic form that we can share across all datasets, which is another step after the datasets have been split.

**Comment by cpuhrsch** [comment]
Yes, the datasets in ```torchtext.experimental.datasets.raw.text_classification``` are being replaced by individual files.

Indeed BC-breaking doesn't quite make sense for a prototype, but I added it anyway.

**Pull request merged**